File: frontend/gestionale-bertozzi/src/app/models/GestioneCommesse/attivita.ts
import moment from 'moment';

export class Attivita {
    id?: number;
    pianoSviluppoId?: number;
    descrizione?: string;
    tipoInfoDaRegistrare?: string;
    percentualeAvanzamento?: number;
    completata?: boolean;
    dataRiferimento?: moment.Moment;
    lettera?: string;
    ordine?: number;
    utenteCreazione?: string;
    utenteModifica?: string;

    static map(attivita: any): Attivita {
        let a = Object.assign(new Attivita(), attivita) as Attivita;
        
        // Converti le stringhe di data in oggetti moment
        if (attivita.dataRiferimento) {
            a.dataRiferimento = moment(attivita.dataRiferimento);
        }
        
        return a;
    }

    static mapArray(arr: any[]): Attivita[] {
        return (arr || []).map(a => Attivita.map(a));
    }
}

File: backend/NemesiLIB/Model/GestioneCommesse/Attivita.cs
using NemesiCOMMONS.Models;
using System;

namespace NemesiLIB.Model.GestioneCommesse
{
    public class Attivita : IAuditable
    {
        public int Id { get; set; }
        public int PianoSviluppoId { get; set; }
        public string Descrizione { get; set; }
        public string TipoInfoDaRegistrare { get; set; }
        public int PercentualeAvanzamento { get; set; }
        public bool Completata { get; set; }
        public DateTime? DataRiferimento { get; set; }
        public string? Lettera { get; set; }
        public int Ordine { get; set; }
        public DateTime DataCreazione { get; set; }
        public DateTime? DataModifica { get; set; }
        public string? UtenteCreazione { get; set; }
        public string? UtenteModifica { get; set; }
    }
}